Add investigation for STOPSIGNAL instruction using kind and podman on Fedora 35

avisiedo opened the pull request:
- Add Dockerfile to build container for the investigation which contains STOPSIGNAL instruction.
- Script for trap several signals.
- Pod configuration.
- kustomize configuration.
- Makefile for automating the steps.
- Documentation at `README.md` file.

Commit:
Add investigation for SIGSTOP instruction when using kubernetes and podman

Commit:
Add investigation to README.md index

tiran commented on incubator/013-signalstop/Dockerfile:
```diff
@@ -0,0 +1,4 @@
+FROM quay.io/fedora/fedora:35
+COPY demo-signal.sh /demo-signal.sh
+STOPSIGNAL SIGINT
+ENTRYPOINT ["/demo-signal.sh"]
```
* https://docs.docker.com/engine/reference/builder/#entrypoint
* https://docs.docker.com/engine/reference/builder/#cmd

tiran replied:
```suggestion
CMD ["/demo-signal.sh"]
```

avisiedo replied:
Thanks! I will exchange the entrypoint by a cmd instruction.

tiran commented on incubator/013-signalstop/demo-signal.sh:
```diff
@@ -0,0 +1,13 @@
+trap trap_sigusr1 SIGUSR1
+trap trap_sigrtmin3 "SIGRTMIN+3"
+
+while true; do sleep 1; done
```
```suggestion
while true; do
    echo -n "."
    sleep 1
done
```

tiran commented on incubator/013-signalstop/demo-signal.sh:
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+function trap_sigint { echo "Exiting by SIGINT";  exit 0; }
```
```suggestion
function trap_sigint { echo -e "\nExiting by SIGINT" >&2;  exit 0; }
```

avisiedo commented on incubator/013-signalstop/kustomization.yaml:
Clean-up this commented section

avisiedo committed:
Give feedback that the process is running

Print out a '.' for giving a feedback that the process is running and responsive.

Co-authored-by: Christian Heimes <christian@python.org>

avisiedo committed:
Improve user experience when running the container directly

Exchange ENTRYPOINT by CMD so that the user experience is improved when running directly the container from the CLI.

Co-authored-by: Christian Heimes <christian@python.org>

avisiedo commented on incubator/013-signalstop/demo-signal.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+function trap_sigint { echo "Exiting by SIGINT";  exit 0; }
+function trap_sigterm { echo "Exiting by SIGTERM"; exit 0; }
+function trap_sigusr1 { echo "Exiting by SIGUSR1"; exit 0; }
+function trap_sigrtmin3 { echo "Exiting by SIGRTMIN+3"; exit 0; }
```
```suggestion
function trap_signal {
  local signal="$1"
  echo -e "\nExiting by ${signal}" >&2
  exit 0
}

for signal in SIGINT SIGTERM SIGUSR1 "SIGRTMIN+3"
do
  trap "trap_signal '${signal}'" "${signal}"
done
```

I like the suggestion and I add a bit more for removing code duplication that I included initially.

avisiedo commented on incubator/013-signalstop/demo-signal.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+function trap_sigint { echo "Exiting by SIGINT";  exit 0; }
+function trap_sigterm { echo "Exiting by SIGTERM"; exit 0; }
+function trap_sigusr1 { echo "Exiting by SIGUSR1"; exit 0; }
+function trap_sigrtmin3 { echo "Exiting by SIGRTMIN+3"; exit 0; }
+
+trap trap_sigint SIGINT
+trap trap_sigterm SIGTERM
+trap trap_sigusr1 SIGUSR1
+trap trap_sigrtmin3 "SIGRTMIN+3"
```
```suggestion
function trap_signal {
  local signal="$1"
  echo -e "\nExiting by ${signal}" >&2
  exit 0
}

for signal in SIGINT SIGTERM SIGUSR1 "SIGRTMIN+3"
do
  trap "trap_signal '${signal}'" "${signal}"
done
```

I like the suggestion and I add a bit more changes for removing duplicated code I included at the beginning. Thanks @tiran

avisiedo committed:
Improve exit message and remove duplicated code

Add @tiran suggestion for redirecting exit message to stderr and remove initial duplicated code.

avisiedo committed:
Clean up commented attributes
